modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -0,0 +1,513 @@
+  mesh0->stitch_surfaces(SLICE_BEGIN, SLICE_END, TOLERANCE, true);
+
+  if (!_has_rings && !_has_ducts && _background_intervals == 1)
+    MooseMesh::changeBoundaryId(*mesh0, 1, OUTER_SIDESET_ID, false);
+
+  if (!_is_general_polygon)
+  {
+    auto mesh0_dup = mesh0->clone();
+    _azimuthal_angle_meta = azimuthal_angles_collector(
+        dynamic_pointer_cast<ReplicatedMesh>(mesh0_dup), -180.0, 180.0, ANGLE_DEGREE);
+    _pattern_pitch_meta = _pitch;
+  }
+
+  // Move nodes on the external boundary for force uniform spacing.
+  if (_uniform_mesh_on_sides)
+  {
+    const Real angle_tol = 1.0e-5;
+    std::vector<std::pair<Real, unsigned int>> node_azi_list;
+    MeshTools::Modification::rotate(*mesh0, -(270.0 - 360.0 / (Real)_num_sides), 0.0, 0.0);
+    std::vector<std::tuple<dof_id_type, unsigned short int, boundary_id_type>> side_list =
+        mesh0->get_boundary_info().build_side_list();
+    mesh0->get_boundary_info().build_node_list_from_side_list();
+    std::vector<std::tuple<dof_id_type, boundary_id_type>> node_list =
+        mesh0->get_boundary_info().build_node_list();
+
+    for (unsigned int i = 0; i < node_list.size(); ++i)
+    {
+      if (std::get<1>(node_list[i]) == OUTER_SIDESET_ID)
+      {
+        node_azi_list.push_back(
+            std::make_pair(atan2((*mesh0->node_ptr(std::get<0>(node_list[i])))(1),
+                                 (*mesh0->node_ptr(std::get<0>(node_list[i])))(0)) *
+                               180.0 / M_PI,
+                           std::get<0>(node_list[i])));
+        if (std::abs(node_azi_list.back().first + 180.0) <= angle_tol)
+          node_azi_list.back().first = 180.0;
+      }
+    }
+    std::sort(node_azi_list.begin(), node_azi_list.end());
+    for (unsigned int i = 0; i < _num_sides; i++)
+    {
+      for (unsigned int j = 1; j <= _num_sectors_per_side[i]; j++)
+      {
+        Real azi_corr_tmp = atan2((Real)j * 2.0 / (Real)_num_sectors_per_side[i] - 1.0,
+                                  1.0 / std::tan(M_PI / (Real)_num_sides));
+        Real x_tmp = _pitch / 2.0;
+        Real y_tmp = x_tmp * std::tan(azi_corr_tmp);
+        node_coord_rotate(&x_tmp,
+                          &y_tmp,
+                          (Real)i * 360.0 / (Real)_num_sides - (180.0 - 180.0 / (Real)_num_sides));
+        Point p_tmp = Point(x_tmp, y_tmp, 0.0);
+        mesh0->add_point(
+            p_tmp,
+            node_azi_list[std::accumulate(
+                              _num_sectors_per_side.begin(), _num_sectors_per_side.begin() + i, 0) +
+                          j - 1]
+                .second);
+      }
+    }
+    MeshTools::Modification::rotate(*mesh0, (270.0 - 360.0 / (Real)_num_sides), 0.0, 0.0);
+  }
+
+  _background_intervals_meta = _background_intervals;
+
+  if (!_has_ducts && _sides_to_adapt.empty())
+  {
+    LaplaceMeshSmoother lms(*mesh0);
+    lms.smooth(_smoothing_max_it);
+  }
+
+  auto mesh1 = dynamic_pointer_cast<MeshBase>(mesh0);
```

Comment:
do you need the cast here or can you do it at the return call?
mesh1 and mesh0 arent very descriptive names